JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
+        try {
+            group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
 
-        em.getTransaction().begin();
-        em.remove(project);
-        em.getTransaction().commit();
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
+
+        return group;
     }
 
+    private Group getGroup(int groupId) {
+
+        Group group;
+        try {
+            group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
 
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
 
-    public Project getProject(String projectName, int groupId) {
+        return group;
+    }
+
+
+
+
+    public String getHash(int projectId) throws NoSuchAlgorithmException, UnsupportedEncodingException {
+
+        Integer projId = new Integer(projectId);
+        String prId = projId.toString();
+        int res = prId.hashCode();
+        return String.valueOf(res);
+
+    }
+
+
+    public Project getProject(String projectHash) {
 
         Project project;
 
-        project = (Project)em.createQuery("SELECT p FROM Project p WHERE p.projectName =:projectname and p.group.groupId =:groupid")
-                .setParameter("projectname", projectName)
-                .setParameter("groupid", groupId)
-                .getSingleResult();
+        try {
+            project = (Project) em.createQuery("SELECT p FROM Project p WHERE p.hash.hash =:projecthash")
+                    .setParameter("projecthash", projectHash)
+                    .getSingleResult();
+        } catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
 
         return project;
 
     }
 
-    private Group getGroup(int groupId) {
+    public Project getProject(String projectName, String groupName) {
+
+        Project p;
+        try {
+            p = (Project) em.createQuery("SELECT p FROM Project p WHERE p.projectName =:projectname AND p.group.groupName =:groupname")
+                    .setParameter("projectname", projectName)
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        }catch (NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
+        return p;
+    }
 
-        Group group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
-                .setParameter("groupid", groupId)
-                .getSingleResult();
+    public Project getProjectByName(String projectName) {
```

Comment:
getProject(...) method names look random.
Best practice here would be refusing from overload and typing getProjectByXxxYyyZzz() in all cases